extensions/librdkafka/ConsumeKafka.cpp

```diff
@@ -0,0 +1,553 @@
+  setKafkaConfigurationField(conf_.get(), "isolation.level", honor_transactions_ ? "read_committed" : "read_uncommitted");
+  setKafkaConfigurationField(conf_.get(), "group.id", group_id_);
+  setKafkaConfigurationField(conf_.get(), "session.timeout.ms", std::to_string(session_timeout_milliseconds_.count()));
+  setKafkaConfigurationField(conf_.get(), "max.poll.interval.ms", "600000");  // Twice the default, arbitrarily chosen
+
+  // This is a librdkafka option, but the communication timeout is also specified in each of the
+  // relevant API calls. Could be redundant, but it probably does not hurt to set this
+  setKafkaConfigurationField(conf_.get(), "metadata.request.timeout.ms", std::to_string(METADATA_COMMUNICATIONS_TIMEOUT_MS));
+
+  extend_config_from_dynamic_properties(context);
+
+  std::array<char, 512U> errstr{};
+  consumer_ = { rd_kafka_new(RD_KAFKA_CONSUMER, conf_.release(), errstr.data(), errstr.size()), utils::rd_kafka_consumer_deleter() };
+  if (consumer_ == nullptr) {
+    const std::string error_msg { errstr.begin(), errstr.end() };
+    throw Exception(PROCESS_SCHEDULE_EXCEPTION, "Failed to create Kafka consumer %s" + error_msg);
+  }
+
+  create_topic_partition_list();
+
+  // Changing the partition list should happen only as part as the initialization of offsets
+  // a function like `rd_kafka_position()` might have unexpected effects
+  // for instance when a consumer gets assigned a partition it used to
+  // consume at an earlier rebalance.
+  //
+  // As far as I understand, instead of rd_kafka_position() an rd_kafka_committed() call if preferred here,
+  // as it properly fetches offsets from the broker
+  if (RD_KAFKA_RESP_ERR_NO_ERROR != rd_kafka_committed(consumer_.get(), kf_topic_partition_list_.get(), METADATA_COMMUNICATIONS_TIMEOUT_MS)) {
+    logger_ -> log_error("Retrieving committed offsets for topics+partitions failed.");
+  }
+
+  rd_kafka_resp_err_t poll_set_consumer_response = rd_kafka_poll_set_consumer(consumer_.get());
+  if (RD_KAFKA_RESP_ERR_NO_ERROR != poll_set_consumer_response) {
+    logger_->log_error("rd_kafka_poll_set_consumer error %d: %s", poll_set_consumer_response, rd_kafka_err2str(poll_set_consumer_response));
+  }
+
+  // There is no rd_kafka_seek alternative for rd_kafka_topic_partition_list_t, only rd_kafka_topic_t
+  // rd_kafka_topic_partition_list_set_offset should reset the offsets to the latest (or whatever is set in the config),
+  // Also, rd_kafka_committed should also fetch and set latest the latest offset
+  // In reality, neither of them seem to work (not even with calling rd_kafka_position())
+  logger_->log_info("Resetting offset manually.");
+  while (true) {
+    std::unique_ptr<rd_kafka_message_t, utils::rd_kafka_message_deleter>
+        message_wrapper{ rd_kafka_consumer_poll(consumer_.get(), max_poll_time_milliseconds_.count()), utils::rd_kafka_message_deleter() };
+
+    if (!message_wrapper || RD_KAFKA_RESP_ERR_NO_ERROR != message_wrapper->err) {
```

Comment:
It is a strange case. We only fetch these messages because the offset reset of the kafka API does not seem to always properly align with the latest offset. As we are discarding these messages either way, I think logging their validity might be misleading.